src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -183,32 +179,44 @@ export function WrapCurrency({ inputCurrency, outputCurrency, gasUseEstimateUSD
 export interface TradeProps extends GasEstimateProps {
   trade: InterfaceTrade
   outputUSDC?: CurrencyAmount<Currency>
-  impact?: PriceImpact
 }
 
-export function Trade({ trade, outputUSDC, impact, gasUseEstimateUSD }: TradeProps) {
+export function Trade({ trade, outputUSDC, gasUseEstimateUSD }: TradeProps) {
   return (
     <>
       <Caption
-        icon={impact?.warning ? AlertTriangle : Info}
+        icon={Info}
         caption={<Price trade={trade} outputUSDC={outputUSDC} />}
         tooltip={{
+          content: <RoutingDiagram trade={trade} />,
+        }}
+      />
+      <GasEstimate gasUseEstimateUSD={gasUseEstimateUSD} />
+    </>
+  )
+}
+
+type PriceImpactWarningProps = Omit<TradeProps, 'gasUseEstimateUSD'> & { impact: PriceImpact }
```

Comment:
nit: It'd also be appropriate here to combine types everywhere, eg:

    export interface TradeProps {...}
    export function Trade({...}: TradeProps & GasEstimateProps) {
    export interface PriceImpactProps extends TradeProps {...}

I'd prefer this to using Omit, because it's less complex overall. wdyt?